src/SymbolTable.ts

```diff
@@ -51,7 +52,9 @@ export class SymbolTable {
     private parentStack: SymbolTable[] = [];
 
     public pushParent(parent?: SymbolTable) {
-        this.parentStack.unshift(parent);
+        if (parent) {
+            this.parentStack.unshift(parent);
+        }
```

Comment:
Remove this. Although pushing undefined parents is not good, the calling code might plan on calling pop later, so we need to keep the undefined in the list so we can pop it.